src/components/ClientPicker/index.jsx

```diff
@@ -26,89 +26,112 @@ import {
-
-import PrivateClientSettings from './PrivateClientSettings';
+} from "../../actions/clientActions";
 
-const propTypes = {
-  client: PropTypes.shape({}).isRequired,
-  network: PropTypes.string.isRequired,
-  setUrl: PropTypes.func.isRequired,
-  setPassword: PropTypes.func.isRequired,
-  setPasswordError: PropTypes.func.isRequired,
-  setType: PropTypes.func.isRequired,
-  setUsername: PropTypes.func.isRequired,
-  setUsernameError: PropTypes.func.isRequired,
-};
+import PrivateClientSettings from "./PrivateClientSettings";
 
 class ClientPicker extends React.Component {
+  static validatePassword() {
+    return "";
+  }
+
+  static validateUsername() {
+    return "";
+  }
+
+  static validateUrl(host) {
+    const validhost = /^http(s)?:\/\/[^\s]+$/.exec(host);
+    if (!validhost) return "Must be a valid URL.";
+    return "";
+  }
+
+  static propTypes = {
+    client: PropTypes.shape({
+      type: PropTypes.string.isRequired,
+    }).isRequired,
+    network: PropTypes.string.isRequired,
+    privateNotes: PropTypes.node,
+    setUrl: PropTypes.func.isRequired,
+    urlError: PropTypes.string,
+    setUrlError: PropTypes.func.isRequired,
+    setPassword: PropTypes.func.isRequired,
+    passwordError: PropTypes.string,
+    setPasswordError: PropTypes.func.isRequired,
+    setType: PropTypes.func.isRequired,
+    usernameError: PropTypes.string,
+    setUsername: PropTypes.func.isRequired,
+    setUsernameError: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    urlError: "",
+    usernameError: "",
+    passwordError: "",
+    privateNotes: React.createElement("span"),
+  };
+
   state = {
-    url_edited: false,
+    urlEdited: false,
     connectError: "",
-    connectSuccess: false
-  }
+    connectSuccess: false,
+  };
 
   handleTypeChange = (event) => {
     const { setType, network, setUrl } = this.props;
-    const type = event.target.value
-    if (type === 'private' && !this.state.url_edited) {
-      setUrl(`http://localhost:${network === 'mainnet' ? 8332 : 18332}`)
+    const { urlEdited } = this.state;
+
+    const type = event.target.value;
+    if (type === "private" && !urlEdited) {
+      setUrl(`http://localhost:${network === "mainnet" ? 8332 : 18332}`);
     }
     setType(type);
-  }
+  };
 
   handleUrlChange = (event) => {
     const { setUrl, setUrlError } = this.props;
+    const { urlEdited } = this.state;
     const url = event.target.value;
-    const error = this.validateUrl(url);
-    if (!this.state.url_edited && !error) this.setState({ url_edited: true });
+    const error = ClientPicker.validateUrl(url);
+    if (!urlEdited && !error) this.setState({ urlEdited: true });
     setUrl(url);
     setUrlError(error);
   };
 
   handleUsernameChange = (event) => {
     const { setUsername, setUsernameError } = this.props;
     const username = event.target.value;
-    const error = this.validateUsername(username);
+    const error = ClientPicker.validateUsername(username);
```

Comment:
I manually made these changes because of [`class-methods-use-this`](https://eslint.org/docs/rules/class-methods-use-this).  When I made these changes I also searched for and updated the usages of these methods.